### ISSUE TITLE
WebGL Error when using multiple lights

### ISSUE THREAD

**Issue by yeemachine** [pixiv/three-vrm#794]
A WebGL error seems to occur when trying to add a PointLight to the scene

```
THREE.WebGLProgram: shader error:  1282 35715 false gl.getProgramInfoLog invalid shaders  THREE.WebGLShader: gl.getShaderInfoLog() fragment
ERROR: 0:1212: 'atten' : redefinition
ERROR: 0:1218: 'shadow' : redefinition
ERROR: 0:1219: 'lightIntensity' : redefinition
ERROR: 0:1220: 'lighting' : redefinition
1: #version 300 es
2: #define varying in
3: out highp vec4 pc_fragColor;
4: #define gl_FragColor pc_fragColor
5: #define gl_FragDepthEXT gl_FragDepth
6: #define texture2D texture
7: #define textureCube texture
8: #define texture2DProj textureProj
9: #define texture2DLodEXT textureLod
10: #define texture2DProjLodEXT textureProjLod
11: #define textureCubeLodEXT textureLod
12: #define texture2DGradEXT textureGrad
13: #define texture2DProjGradEXT textureProjGrad
14: #define textureCubeGradEXT textureGrad
15: precision highp float;
16: precision highp int;
17: #define HIGH_PRECISION
18: #define SHADER_NAME ShaderMaterial
19: #define BLENDMODE_TRANSPARENT true
20: #define MTOON_USE_UV true
21: #define USE_SHADETEXTURE true
22: #define USE_SPHEREADD true
23: #define OUTLINE_COLOR_MIXED true
24: #define GAMMA_FACTOR 2
25: #define USE_MAP
26: #define USE_EMISSIVEMAP
27: #define USE_BUMPMAP
28: #define USE_NORMALMAP
29: #define TANGENTSPACE_NORMALMAP
30: #define USE_UV
31: uniform mat4 viewMatrix;
32: uniform vec3 cameraPosition;
33: uniform bool isOrthographic;
34: 
35: vec4 LinearToLinear( in vec4 value ) {
36: 	return value;
37: }
38: vec4 GammaToLinear( in vec4 value, in float gammaFactor ) {
39: 	return vec4( pow( value.rgb, vec3( gammaFactor ) ), value.a );
40: }
41: vec4 LinearToGamma( in vec4 value, in float gammaFactor ) {
42: 	return vec4( pow( value.rgb, vec3( 1.0 / gammaFactor ) ), value.a );
43: }
44: vec4 sRGBToLinear( in vec4 value ) {
45: 	return vec4( mix( pow( value.rgb * 0.9478672986 + vec3( 0.0521327014 ), vec3( 2.4 ) ), value.rgb * 0.0773993808, vec3( lessThanEqual( value.rgb, vec3( 0.04045 ) ) ) ), value.a );
46: }
47: vec4 LinearTosRGB( in vec4 value ) {
48: 	return vec4( mix( pow( value.rgb, vec3( 0.41666 ) ) * 1.055 - vec3( 0.055 ), value.rgb * 12.92, vec3( lessThanEqual( value.rgb, vec3( 0.0031308 ) ) ) ), value.a );
49: }
50: vec4 RGBEToLinear( in vec4 value ) {
51: 	return vec4( value.rgb * exp2( value.a * 255.0 - 128.0 ), 1.0 );
52: }
53: vec4 LinearToRGBE( in vec4 value ) {
54: 	float maxComponent = max( max( value.r, value.g ), value.b );
55: 	float fExp = clamp( ceil( log2( maxComponent ) ), -128.0, 127.0 );
56: 	return vec4( value.rgb / exp2( fExp ), ( fExp + 128.0 ) / 255.0 );
57: }
58: vec4 RGBMToLinear( in vec4 value, in float maxRange ) {
59: 	return vec4( value.rgb * value.a * maxRange, 1.0 );
60: }
61: vec4 LinearToRGBM( in vec4 value, in float maxRange ) {
62: 	float maxRGB = max( value.r, max( value.g, value.b ) );
63: 	float M = clamp( maxRGB / maxRange, 0.0, 1.0 );
64: 	M = ceil( M * 255.0 ) / 255.0;
65: 	return vec4( value.rgb / ( M * maxRange ), M );
66: }
67: vec4 RGBDToLinear( in vec4 value, in float maxRange ) {
68: 	return vec4( value.rgb * ( ( maxRange / 255.0 ) / value.a ), 1.0 );
69: }
70: vec4 LinearToRGBD( in vec4 value, in float maxRange ) {
71: 	float maxRGB = max( value.r, max( value.g, value.b ) );
72: 	float D = max( maxRange / maxRGB, 1.0 );
73: 	D = clamp( floor( D ) / 255.0, 0.0, 1.0 );
74: 	return vec4( value.rgb * ( D * ( 255.0 / maxRange ) ), D );
75: }
76: const mat3 cLogLuvM = mat3( 0.2209, 0.3390, 0.4184, 0.1138, 0.6780, 0.7319, 0.0102, 0.1130, 0.2969 );
77: vec4 LinearToLogLuv( in vec4 value ) {
78: 	vec3 Xp_Y_XYZp = cLogLuvM * value.rgb;
79: 	Xp_Y_XYZp = max( Xp_Y_XYZp, vec3( 1e-6, 1e-6, 1e-6 ) );
80: 	vec4 vResult;
81: 	vResult.xy = Xp_Y_XYZp.xy / Xp_Y_XYZp.z;
82: 	float Le = 2.0 * log2(Xp_Y_XYZp.y) + 127.0;
83: 	vResult.w = fract( Le );
84: 	vResult.z = ( Le - ( floor( vResult.w * 255.0 ) ) / 255.0 ) / 255.0;
85: 	return vResult;
86: }
87: const mat3 cLogLuvInverseM = mat3( 6.0014, -2.7008, -1.7996, -1.3320, 3.1029, -5.7721, 0.3008, -1.0882, 5.6268 );
88: vec4 LogLuvToLinear( in vec4 value ) {
89: 	float Le = value.z * 255.0 + value.w;
90: 	vec3 Xp_Y_XYZp;
91: 	Xp_Y_XYZp.y = exp2( ( Le - 127.0 ) / 2.0 );
92: 	Xp_Y_XYZp.z = Xp_Y_XYZp.y / value.y;
93: 	Xp_Y_XYZp.x = value.x * Xp_Y_XYZp.z;
94: 	vec3 vRGB = cLogLuvInverseM * Xp_Y_XYZp.rgb;
95: 	return vec4( max( vRGB, 0.0 ), 1.0 );
96: }
97: vec4 mapTexelToLinear( vec4 value ) { return LinearToLinear( value ); }
98: vec4 emissiveMapTexelToLinear( vec4 value ) { return LinearToLinear( value ); }
99: vec4 linearToOutputTexel( vec4 value ) { return LinearToLinear( value ); }
100: 
101: vec4 shadeTextureTexelToLinear( vec4 value ) { return LinearToLinear( value ); }
102: vec4 sphereAddTexelToLinear( vec4 value ) { return LinearToLinear( value ); }
103: // #define PHONG
104: 
105: #ifdef BLENDMODE_CUTOUT
106:   uniform float cutoff;
107: #endif
108: 
109: uniform vec3 color;
110: uniform float colorAlpha;
111: uniform vec3 shadeColor;
112: #ifdef USE_SHADETEXTURE
113:   uniform sampler2D shadeTexture;
114: #endif
115: 
116: uniform float receiveShadowRate;
117: #ifdef USE_RECEIVESHADOWTEXTURE
```

**Comment by 0b5vr** [comment]
Thank you for the report! I will investigate.

**Comment by 0b5vr** [comment]
Reproduced, but it's not when we are using PointLight but using multiple any lights.
Going to fix this :bow:

**Comment by 0b5vr** [comment]
@yeemachine it should be fixed @ `v0.6.5` I just released :bow:

**Comment by yeemachine** [comment]
@0b5vr Thanks! can confirm it is fixed!
